File: data_structures/python/Queue.py
class Queue():
    def __init__(self):
        self.queue = []

    # adds item to the end of the queue
    def enqueue(self, data):
        self.queue.append(data)

    # removes item from front of the queue
    def dequeue(self, data):
        self.queue.pop(0)

    def front(self):
        print(self.queue[0])

    def rear(self):
        print(self.queue[-1])


queue = Queue()
queue.enqueue(1)

queue.front()
queue.rear()

File: data_structures/java/Queue.java
// implementation of a queue class using a LinkedList under the hood

public class Queue {
    public LinkedList queue;

    public Queue() {
        this.queue = new LinkedList();
    }

    // adds an item to the end of the queue
    public void enqueue(int item) {
        this.queue.addToTail(item);
    }

    // removes an item from the front of the queue
    public int dequeue() {
        return this.queue.removeFromHead();
    }

    // returns the item's value from the front of the queue
    public int peek() {
        return this.queue.getHead().getData();
    }

    public boolean isEmpty() {
        return this.queue.getHead() == null;
    }

    public void print() {
        this.queue.printList();
    }



}
